pkg/operator/event_handler.go

```diff
@@ -57,15 +56,31 @@ func (e *EventHandler) OnAdd(obj interface{}, _ bool) {
 }
 
 func (e *EventHandler) OnUpdate(old, cur interface{}) {
-	if old.(metav1.Object).GetResourceVersion() == cur.(metav1.Object).GetResourceVersion() {
+	oldMeta, ok := e.accessor.ObjectMetadata(old)
+	if !ok {
+		return
+	}
+	curMeta, ok := e.accessor.ObjectMetadata(cur)
+	if !ok {
 		return
 	}
 
-	if o, ok := e.accessor.ObjectMetadata(cur); ok {
-		e.logger.Debug(fmt.Sprintf("%s updated", e.objName))
-		e.metrics.TriggerByCounter(e.objName, UpdateEvent)
-		e.enqueueFunc(o.GetNamespace())
+	// Generation value is 0 for ConfigMaps and Secrets.
+	if curMeta.GetGeneration() == 0 {
+		if oldMeta.GetResourceVersion() == curMeta.GetResourceVersion() {
+			return
+		}
+	} else {
+		if reflect.DeepEqual(oldMeta.GetLabels(), curMeta.GetLabels()) &&
+			reflect.DeepEqual(oldMeta.GetAnnotations(), curMeta.GetAnnotations()) &&
+			oldMeta.GetGeneration() == curMeta.GetGeneration() {
+			return
+		}
 	}
```

Comment:
Can we reduce the nesting here? It would make sense to abstract the comparisons away into functions if you can see yourself using them again, or just to make the code a bit easier to read.
```suggestion
	if curMeta.GetGeneration == 0 && !hasResourceVersionChanged(oldMeta, curMeta) {
	return 
	} else if  !hasObjMetaChanged(oldMeta, curMeta) {
	return 
	}
```
We can move the entire nest into one helper like `shouldEnqueueObj()`, or have a couple of helpers like above, which state clearly what they do.

Reply:
i dont think this will be used anywhere else, but for better readiblity we can do this